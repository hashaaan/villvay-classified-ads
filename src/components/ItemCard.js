import {
  Box,
  Card,
  CardContent,
  CardActionArea,
  Typography,
  IconButton,
} from "@mui/material";
import { FavoriteBorder, Favorite } from "@mui/icons-material";

const ItemCard = ({ item, onClick, onFavorite }) => {
  return (
    <Card>
      <Box
        sx={{
          display: "block",
          backgroundImage: `url(${item.imageUrl})`,
          backgroundSize: "cover",
          backgroundPosition: "center",
          height: 170,
        }}
      >
        <IconButton
          color="primary"
          aria-label="add to favorites"
          sx={[
            {
              backgroundColor: "#FFF",
              color: item.isFavourite ? "#1976d2" : "#969696",
              margin: "10px",
              height: "40px",
              float: "right",
            },
            {
              "&:hover": {
                backgroundColor: "#FFF",
                color: "#1976d2",
              },
            },
          ]}
          onClick={() => onFavorite(item)}
        >
          {item.isFavourite ? <Favorite /> : <FavoriteBorder />}
        </IconButton>
      </Box>
      <CardActionArea onClick={onClick}>
        <CardContent>
          <Typography
            sx={{ fontSize: 10, fontWeight: "600", letterSpacing: 1 }}
            gutterBottom
          >
            {item.category.toUpperCase()}
          </Typography>
          <Typography
            gutterBottom
            sx={{ fontWeight: "500" }}
            variant="h5"
            component="div"
          >
            {item.name}
          </Typography>
          <Typography variant="body2" color="text.secondary">
            {item.description}
          </Typography>
        </CardContent>
      </CardActionArea>
    </Card>
  );
};

export default ItemCard;
