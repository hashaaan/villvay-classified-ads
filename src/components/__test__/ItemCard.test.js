import React from "react";
import ReactDom from "react-dom";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

import ItemCard from "../ItemCard";

const mockedUsedNavigate = jest.fn();

const mokedItem = {
  id: 1,
  name: "Lorem ipsum dolor sit amet",
  category: "housing",
  imageUrl: "/static/images/07.png",
  description:
    "consectetur adipiscing elit. Morbi eleifend risus quis quam euismod, eget aliquam nulla elementum. Nullam lacus nisi, pretium in viverra a, consectetur id mauris. Maecenas luctus risus quis vehicula cursus...",
  isFavourite: false,
};

afterEach(cleanup);

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

it("renders item card without crashing", () => {
  const div = document.createElement("div");
  ReactDom.render(
    <ItemCard item={mokedItem} onClick={() => {}} onFavoriteClick={() => {}} />,
    div
  );
});

it("renders favourite button correctly", () => {
  const { getByTestId } = render(
    <ItemCard item={mokedItem} onClick={() => {}} onFavoriteClick={() => {}} />
  );
  getByTestId("favourite-button");
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <ItemCard
        item={mokedItem}
        onClick={() => {}}
        onFavoriteClick={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
