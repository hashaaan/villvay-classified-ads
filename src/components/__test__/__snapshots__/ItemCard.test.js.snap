// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiBox-root css-pm9sir"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-gffpc6-MuiButtonBase-root-MuiIconButton-root"
      data-testid="favourite-button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="FavoriteBorderIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
        />
      </svg>
    </button>
  </div>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-11ykory-MuiTypography-root"
      >
        HOUSING
      </p>
      <div
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1er9w5e-MuiTypography-root"
      >
        Lorem ipsum dolor sit amet
      </div>
      <p
        className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        consectetur adipiscing elit. Morbi eleifend risus quis quam euismod, eget aliquam nulla elementum. Nullam lacus nisi, pretium in viverra a, consectetur id mauris. Maecenas luctus risus quis vehicula cursus...
      </p>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
